lib/octopus_serverspec_extensions/type/octopus_deploy_project_group.rb

```diff
@@ -0,0 +1,105 @@
+require 'serverspec/type/base'
+require 'net/http'
+require 'json'
+
+module Serverspec::Type
+  class OctopusDeployProjectGroup < Base
+    @projectgroup = nil
+    @serverUrl = nil
+    @apiKey = nil
+    @serverSupportsSpaces = nil
+    @spaceId = nil
+    @spaceFragment = ""
+
+    def initialize(serverUrl, apiKey, projectgroup_name, spaceID = 'Spaces-1')
+      @name = "Octopus Deploy Project Group #{projectgroup_name}"
+      @runner = Specinfra::Runner
+      @serverUrl = serverUrl
+      @apiKey = apiKey
+      @spaceId = spaceID
+
+      if serverUrl.nil?
+        raise "'serverUrl' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+      if apiKey.nil?
+        raise "'apiKey' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+      if projectgroup_name.nil?
+        raise "'projectgroup_name' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+
+      @serverSupportsSpaces = check_supports_spaces(serverUrl)
+
+      if @serverSupportsSpaces
+        @spaceFragment = "#{@spaceId}/"
+      end
+
+      @projectgroup = get_projectgroup_via_api(serverUrl, apiKey, projectgroup_name)
+    end
+
+    def exists?
+      (!@projectgroup.nil?) && (@projectgroup != [])
+    end
+
+    def has_description?(projectgroup_description)
+      return false if @projectgroup.nil?
+      @projectgroup["Description"] == projectgroup_description
+    end
+
+    def in_space?(space_name)
```

Comment:
Not sure about this... The space name is already passed in the ctor?

Reply:
SpaceID in the constructor, isn't it?

Reply:
Yep, the constructor is SpaceID. I kinda wonder if there should be a SpaceName _or_ SpaceID in the ctor maybe.

Reply:
I mean, you've already passed the space id into the constructor, and all the checks are based on whether the server says that object exists in that space.

Reply:
Yeah, kinda pointless, isn't it?